postgres_composite_types/__init__.py

```diff
@@ -258,10 +259,32 @@ def __init__(cls, name, bases, attrs):
         if name == 'CompositeType':
             return
 
+        cls._capture_descriptors()
+
         # Register the type on the first database connection
         connection_created.connect(receiver=cls.database_connected,
                                    dispatch_uid=cls._meta.db_type)
 
+    def _capture_descriptors(cls):
+        """ Work around for not being able to call contribute_to_class."""
+
+        # Too much to fake to call contribute_to_class directly, but we still
+        # want fields to be able to set custom type descriptors.
+        # So we fake a model instead, and find any descriptors on that
+        attrs = {field_name: field for field_name, field in cls._meta.fields}
+
+        class Meta:
+            app_label = cls.__module__
+
+        attrs['__module__'] = cls.__module__
+        attrs['Meta'] = Meta
+        fake_model = type(cls.__name__ + 'FakeModel', (models.Model,), attrs)
```

Comment:
I've changed it to `_Fake{}Model.format(cls.__name__)`,  it needs to be a unique app_label and model class name combo otherwise Django issues a warning about model reloading. I could go into Django's internals a bit and unload the model once we've extracted what we need, but I figured it was less fragile just to use the unique data we have to construct a class name. (I've also added a code comment about this).